pkg/util/log/logcrash/crash_reporting.go

```diff
@@ -56,8 +57,44 @@ var (
 		settings.ApplicationLevel,
 		"diagnostics.reporting.enabled",
 		"enable reporting diagnostic metrics to cockroach labs",
-		false,
-		settings.WithPublic)
+		true,
+		settings.WithPublic,
+		settings.WithValidateBool(func(sv *settings.Values, b bool) error {
+			// if the user wants to turn on diagnostics, short circuit to no error
+			if b {
+				return nil
+			}
+
+			// attempt to get the license to verify ability to disable reporting
+			licenseSetting, ok, _ := settings.LookupForLocalAccess("enterprise.license", true /* forSystemTenant */)
+			if !ok {
+				log.Warning(context.Background(), "unable to find license configuring diagnostic reporting")
```

Comment:
these are warnings because it's uncertain if on startup whether the value will be set by the time this point in code is hit